crates/nu-cli/src/commands/sort_by.rs

```diff
@@ -135,6 +135,15 @@ pub fn sort(
         } => {
             let should_sort_case_insensitively = insensitive && vec.iter().all(|x| x.is_string());
 
+            if !vec
+                .windows(2)
+                .all(|elem| coerce_compare(&elem[0], &elem[1]).is_ok())
+            {
+                return Err(ShellError::labeled_error("Can't reduce all the values to all the values to the same type",
+                                                        "All these values should have some relation between then such that the shell knows how to compare them",
+                                                        tag));
```

Comment:
A few things:

1. style nit: the indentation is a little hard to follow here, and also puts the values far to the right.
2. You've repeated "all the values" in the messae. I think you want something like "Can't reduce all of the values to the same type"?
3. We generally want small labels (the second parameter), and more detailed messages.

Given those things, I'm thinking something along these lines (feel free to tweak):

```suggestion
                return Err(ShellError::labeled_error(
                    "Not all values can be compared",
                    "not all values compare",
                    tag
                ));
```

The label and message are basically repeated, but that's typical when you don't have any more specific information to include in the message itself. What do you think?

**Bonus round:**

If you're up for it, what if we could include the specific types that don't compare? Knowing the types that aren't comparing gives the user a little more information to work with to resolve the issue.

`coerce_compare` will return an error that is a pair of `str` values naming the types that can't be compared. What if, instead of `all`, you had an iterator that return `Option<(usize, str, str)>`, where the `usize` is the index where things didn't compare, and the two `str` values are the names of the two types. If the result was `Some(...)`, you could produce a much more detailed message then

Happy to give you some guidance on how you could approach this, if desired 🙂

Reply:
I'm going to change the labes to what you suggested and i'm definitely going to try to give the index and type of the values that can't be coerced

Reply:
Sounds good. Reach out to me on discourse if you'd like any tips/help. I'm thinking the end result could be a message along the lines

> Not all values can be compared (<type1> cannot compare against <type2>).

If you'd like, you can make the style/message tweaks in this PR, and you can try the "bonus" in a separate PR. That way we can get your contribution in, which always feels good 🙂